tsdb/exemplar.go

```diff
@@ -0,0 +1,220 @@
+// Copyright 2020 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tsdb
+
+import (
+	"context"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/pkg/exemplar"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+type exemplarMetrics struct {
+	outOfOrderExemplars prometheus.Counter
+}
+
+func newExemplarMetrics(r prometheus.Registerer) *exemplarMetrics {
+	m := &exemplarMetrics{
+		outOfOrderExemplars: prometheus.NewCounter(prometheus.CounterOpts{
+			Name: "prometheus_exemplar_out_of_order_exemplars_total",
```

Comment:
as this is in the tsdb package I would expect to have tsdb in the name ofthe metric
```suggestion
			Name: "prometheus_tsdb_exemplars_out_of_order_total",
```